plugins/helper.js

```diff
@@ -0,0 +1,253 @@
+                    return resolve(res.body[0]);
+                }
+
+                return resolve(null);
+            }
+
+            return reject(err);
+        });
+    });
+}
+
+/**
+ *
+ * @param {String} apiUri
+ * @param {Object} eventConfig
+ * @param {Object} buildEvent
+ * @param {Function} retryStrategyFn
+ */
+async function createBuildEvent(apiUri, eventConfig, buildEvent, retryStrategyFn) {
+    const { redisInstance, buildId } = eventConfig;
+    const buildConfig = await redisInstance.hget(`${queuePrefix}buildConfigs`, buildId)
+        .then(JSON.parse);
+
+    return new Promise((resolve, reject) => {
+        requestretry({
+            json: true,
+            method: 'POST',
+            uri: `${apiUri}/v4/events`,
+            headers: {
+                Authorization: `Bearer ${buildConfig.token}`,
+                'Content-Type': 'application/json'
+            },
+            body: buildEvent,
+            maxAttempts: RETRY_LIMIT,
+            retryDelay: RETRY_DELAY * 1000, // in ms
+            retryStrategy: retryStrategyFn
+        }, (err, res) => {
+            if (!err && res.statusCode === 201) {
+                return resolve(res);
+            }
+            if (res.statusCode !== 201) {
+                return reject(JSON.stringify(res.body));
+            }
+
+            return reject(err);
+        });
+    });
+}
+
+/**
+ *
+ * @param {String} apiUri
+ * @param {String} pipelineId
+ * @param {Function} retryStrategyFn
+ */
+async function getPipelineAdmin(requestConfig, apiUri, pipelineId, retryStrategyFn) {
+    const { redisInstance, buildId } = requestConfig;
+    const buildConfig = await redisInstance.hget(`${queuePrefix}buildConfigs`, buildId)
+        .then(JSON.parse);
+
+    return new Promise((resolve, reject) => {
+        requestretry({
+            json: true,
+            method: 'GET',
+            uri: `${apiUri}/pipelines/${pipelineId}/admin`,
+            headers: {
+                Authorization: `Bearer ${buildConfig.token}`,
+                'Content-Type': 'application/json'
+            },
+            maxAttempts: RETRY_LIMIT,
+            retryDelay: RETRY_DELAY * 1000, // in ms
+            retryStrategy: retryStrategyFn
+        }, (err, res) => {
+            if (!err && res.statusCode === 201) {
+                return resolve(res);
+            }
+            if (res.statusCode !== 201) {
+                return reject(JSON.stringify(res.body));
+            }
+
+            return reject(err);
+        });
+    });
+}
+
+/**
+ *
+ * @param {String} buildId
+ * @param {String} token
+ * @param {String} status
+ * @param {String} statusMessage
+ * @param {String} apiUri
+ * @param {Object} updateConfig
+ */
+async function updateBuildStatusWithRetry(updateConfig) {
+    const { buildId, token, status, statusMessage, apiUri } = updateConfig;
+    // const buildConfig = await redisInstance.hget(`${queuePrefix}buildConfigs`, buildId)
```

Comment:
commented out code ? Also there there seems to be common options for requestrestry which could be defined in a global constant.